authentication/openshift.go

```diff
@@ -501,7 +504,36 @@ func (a OpenShiftAuthenticator) Middleware() Middleware {
 
 func (a OpenShiftAuthenticator) GRPCMiddleware() grpc.StreamServerInterceptor {
 	return grpc_middleware_auth.StreamServerInterceptor(func(ctx context.Context) (context.Context, error) {
-		return ctx, status.Error(codes.Unimplemented, "internal error")
+		httpReq := http.Request{Header: map[string][]string{}}
+
+		// set authorization header
+		md, ok := metadata.FromIncomingContext(ctx)
+		if !ok {
+			return ctx, status.Error(codes.Internal, "metadata error")
+		}
+		for _, val := range md.Get("Authorization") {
+			httpReq.Header.Add("Authorization", val)
+		}
+
+		// set peer certificates
+		peerContext, ok := peer.FromContext(ctx)
+		if ok {
+			tlsInfo := peerContext.AuthInfo.(credentials.TLSInfo)
```

Comment:
```suggesstion
			tlsInfo, ok := peerContext.AuthInfo.(credentials.TLSInfo)
```
@pavolloffay would it make sense to check the type cast here to prevent pontential problems in the future? or is it covered elsewhere?